frontend/src/api/k8s/inferenceServices.ts

```diff
@@ -133,89 +146,55 @@ export const assembleInferenceService = (
-            ...inferenceService.metadata.annotations,
-            ...annotations,
-          },
-          labels: {
-            ...inferenceService.metadata.labels,
-          },
-        },
-        spec: {
-          predictor: {
-            ...(!isModelMesh && { minReplicas }),
-            ...(!isModelMesh && { maxReplicas }),
-            ...(!isModelMesh && { imagePullSecrets }),
-            model: {
-              modelFormat: {
-                name: format.name,
-                ...(format.version && { version: format.version }),
-              },
-              runtime: servingRuntimeName,
-              ...(uri
-                ? { storageUri: uri }
-                : {
-                    storage: {
-                      key: dataConnectionKey,
-                      path,
-                    },
-                  }),
-              args: splitArgs,
-              env: nonEmptyEnvVars,
-            },
-          },
-        },
-      }
-    : {
-        apiVersion: 'serving.kserve.io/v1beta1',
-        kind: 'InferenceService',
-        metadata: {
-          annotations,
-          labels: {
-            [KnownLabels.DASHBOARD_RESOURCE]: 'true',
-            ...data.labels,
-          },
-          name,
-          namespace: project,
-        },
-        spec: {
-          predictor: {
-            ...(!isModelMesh && { minReplicas }),
-            ...(!isModelMesh && { maxReplicas }),
-            ...(!isModelMesh && { imagePullSecrets }),
-            model: {
-              modelFormat: {
-                name: format.name,
-                ...(format.version && { version: format.version }),
-              },
-              runtime: servingRuntimeName,
-              ...(uri
-                ? { storageUri: uri }
-                : {
-                    storage: {
-                      key: dataConnectionKey,
-                      path,
-                    },
-                  }),
-              args: splitArgs,
-              env: nonEmptyEnvVars,
-            },
-          },
-        },
-      };
+  const labels = { ...updatedInferenceService.metadata.labels, ...data.labels };
+  labels[KnownLabels.DASHBOARD_RESOURCE] = 'true';
+
+  updatedInferenceService.metadata.annotations = annotations;
+  updatedInferenceService.metadata.labels = labels;
 
-  updateInferenceService = applyAuthToInferenceService(
-    updateInferenceService,
+  const spec = { ...updatedInferenceService.spec };
+  const predictor = { ...spec.predictor };
+
+  if (!isModelMesh) {
+    predictor.minReplicas = minReplicas;
+    predictor.maxReplicas = maxReplicas;
+    predictor.imagePullSecrets = imagePullSecrets;
+  }
+
+  const model = { ...predictor.model };
+  model.modelFormat = {
+    name: format.name,
+    ...(format.version && { version: format.version }),
+  };
+  model.runtime = servingRuntimeName;
+
+  if (uri) {
+    model.storageUri = uri;
+    delete model.storage;
+  } else {
+    delete model.storageUri;
```

Comment:
hmmm I don't think this is working actually...

Reply:
Nevermind this is because the ConnectionSection had a bug where selecting a new existing connection didn't clear the URI data